controllers/cf/cf_pod_spec.go

```diff
@@ -0,0 +1,318 @@
+
+The operator may still modify the values you provided,
+but it will not add required sub-fields or fix an invalid value.
+*/
+func SanitizeBasePodSpec(log *zap.SugaredLogger, base *ar.ApicurioRegistrySpecDeploymentPodSpecPreview, current *core.PodSpec, factory *core.PodSpec) (*ar.ApicurioRegistrySpecDeploymentPodSpecPreview, *SanitizeError) {
+	// We are using values from *current* with fields (values):
+	//   - that the user cannot change
+	//   - or are empty by default
+	//   - or there is a CF that handles them
+	// We are using values from *factory* with fields (values):
+	//   - that are not empty by default, so empty fields in the base podSpec do not remove default values
+	//   - and there is not and existing CF that handles them
+	// Otherwise we just pass the base values along, relying on CFs to override things
+
+	base = base.DeepCopy()       // Defensive copy TODO is this needed?
+	current = current.DeepCopy() // Defensive copy TODO is this needed?
+	// affinity
+	if base.Affinity != nil {
+		return nil, NewConfigError("affinity")
+	}
+	base.Affinity = current.Affinity
+	// containers[*]
+	if len(base.Containers) > 1 {
+		return nil, NewConfigError("field containers must contain at most one item")
+
+	} else if len(base.Containers) == 1 {
+		baseContainer := &base.Containers[0]
+		// We have only a single container at the moment,
+		// TODO but in the future we should use the name field.
+		currentContainer := current.Containers[0]
+		factoryContainer := factory.Containers[0]
+
+		// containers[*].args
+		if len(baseContainer.Args) > 0 {
+			return nil, NewConfigError("containers[0].args")
+		}
+		baseContainer.Args = currentContainer.Args
+		// containers[*].command
+		if len(baseContainer.Command) > 0 {
+			return nil, NewConfigError("containers[0].command")
+		}
+		baseContainer.Command = currentContainer.Command
+		// containers[*].env
+		if len(baseContainer.Env) > 0 {
+			return nil, NewConfigError("containers[0].env")
+		}
+		baseContainer.Env = currentContainer.Env
+		// containers[*].image
+		if baseContainer.Image != "" {
+			return nil, NewConfigError("containers[0].image")
+		}
+		baseContainer.Image = currentContainer.Image
+		// containers[*].imagePullPolicy
+		if baseContainer.ImagePullPolicy != "" {
+			return nil, NewConfigError("containers[0].imagePullPolicy")
+		}
+		baseContainer.ImagePullPolicy = currentContainer.ImagePullPolicy
+		// containers[*].livenessProbe
+		if baseContainer.LivenessProbe == nil {
+			baseContainer.LivenessProbe = factoryContainer.LivenessProbe
+		}
+		// containers[*].name
+		if baseContainer.Name != "" {
+			return nil, NewConfigError("containers[0].name")
+		}
+		baseContainer.Name = factoryContainer.Name
+		// containers[*].ports
+		if len(baseContainer.Ports) > 0 {
+			return nil, NewConfigError("containers[0].ports")
+		}
+		baseContainer.Ports = currentContainer.Ports
+		// containers[*].readinessProbe
+		if baseContainer.ReadinessProbe == nil {
```

Comment:
I think we are missing to reconcile the `StartupProbe`.

In addition, to help the review of this code, can I ask to keep a consistent ordering of the fields?
Otherwise is going to be hard to check the "exhaustivity" of this.